CONTRIBUTING.md

```diff
@@ -0,0 +1,346 @@
+ - Make your change and commit it.
+ - Push the branch to your fork.
+ - Open a PR against the upstream repo.
+
+You can find some easy tutorial online such as [this one](https://opensource.com/article/19/7/create-pull-request-github)
+and check out the official [GitHub docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests)
+that contain much more detail.
+
+All development happens on the `main` branch so all PRs should be submitted against that branch.
+Maintainers will take care of backporting if needed.
+
+While bug fixes can first be identified via an "issue" in Github, that is not required.
+It's ok to just open up a PR with the fix, but make sure you include the same information you would have included in an issue - like how to reproduce it.
+
+PRs for new features should include some background on what use cases the new code is trying to address.
+When possible and when it makes sense, try to break-up larger PRs into smaller ones - it's easier to review smaller code changes.
+But only if those smaller ones make sense as stand-alone PRs.
+
+Regardless of the type of PR, all PRs should include:
+* Well-documented code changes, both through comments in the code itself and high-quality commit messages.
+  A commit message should answer *why* a change was made.
+* Additional tests. Ideally, they should fail without your code change applied. A test can be a unit test
+  (which should be added in the same file as the code being tested) or in a more complex suite often found in the
+  `test/` or `tests/` directory in each respective repo.
+  Sometimes it may not be possible to add a useful test (e.g. a race condition that is very hard to trigger),
+  in that case a maintainer can decide to merge without tests.
+* Documentation updates to reflect the changes made in the pull request often found in the `docs/` directory.
+
+Squash your commits into logical pieces of work that might want to be reviewed separate from the rest of the PRs.
+Code changes, test and documentation updates should be part of the same commit as long as they are for the same
+feature/bug fix. Dependency updates are best kept in an individual commit. Totally unrelated changes, i.e.
+fixing typos in a different code part or adding a completely different feature should go into their own PR.
+Often squashing down to just one commit is acceptable since in the end the entire PR will be reviewed anyway.
+When in doubt, ask a maintainer how they prefer it.
+
+When your PR fixes an issue, please note that by including `Fixes: #00000` in the commit description.
+More details on this are below, in the "Describe your changes in Commit Messages" section.
+
+This repository follows a two-ack policy for merges.
+PRs will be approved by an approver listed in [`OWNERS`](OWNERS) file in the root of the repository.
+They will then be merged by a repo owner. Two reviews are required for a pull request to merge.
```

Comment:
Does this apply to this repository?

Reply:
Yes, we use this for all repos. Netavark and Aardvark only have 3 maintainers right now which makes it interesting to try and make work, but it is our goal.